src/commands/infra/init.ts

```diff
@@ -0,0 +1,94 @@
+import child_process from "child_process";
+import commander from "commander";
+import emoji from "node-emoji";
+import { promisify } from "util";
+import { logger } from "../../logger";
+
+/**
+ * Adds the init command to the commander command object
+ *
+ * @param command Commander command object to decorate
+ */
+
+// Check for versions for each executable
+
+let binaries: string[] = ["terraform", "git", "az", "helm"];
+let envVar: string[] = [
+  "ARM_SUBSCRIPTION_ID",
+  "ARM_CLIENT_ID",
+  "ARM_CLIENT_SECRET",
+  "ARM_TENANT_ID"
+];
+
+export const initCommand = (command: commander.Command): void => {
+  command
+    .command("init")
+    .alias("i")
+    .description(
+      "Initialize will verify that all infrastructure deployment prerequisites have been correctly installed."
+    )
+    .action(async opts => {
+      try {
+        if (await validatePrereqs(binaries)) {
+          if (await validateAzure()) {
+            await validateEnvVariables(envVar);
+          }
+        }
+      } catch (err) {
+        logger.error(`Error validating init prerequisites`);
+        logger.error(err);
+      }
+    });
+};
+
+export const validatePrereqs = async (
+  executables: string[]
+): Promise<boolean> => {
+  // Validate executables in PATH
+  for (let i of executables) {
+    try {
+      await promisify(child_process.exec)("which " + i);
```

Comment:
How do we plan to enforce particular versions of these prereqs?

Reply:
Enforcing specific versions of prereqs will come in the next iteration of building the `init` command. We plan on removing the hardcoded binaries in the array and possibly storing versions in a separate file that is specific to a version of the spk tool. This is still a discussion to be had regarding compatibility with specific Bedrock releases as well.